server/src/main/java/org/elasticsearch/action/admin/cluster/health/TransportClusterHealthAction.java

```diff
@@ -133,7 +133,9 @@ protected void onPublicationComplete() {
                     final long timeoutInMillis = Math.max(0, endTimeRelativeMillis - threadPool.relativeTimeInMillis());
                     final TimeValue newTimeout = TimeValue.timeValueMillis(timeoutInMillis);
                     request.timeout(newTimeout);
-                    executeHealth(
+
+                    // Move off of the MasterService thread, back onto the MANAGEMENT thread.
```

Comment:
nit I see you re-use this same comment in the other places you use forkAndExecuteHealth. You could simply add a javadoc to that function with the same comment. It's expected we fork to other executors for such reasons.